packages/react-router-dom/modules/NavLink.js

```diff
@@ -19,7 +20,7 @@ const NavLink = ({
   "aria-current": ariaCurrent,
   ...rest
 }) => {
-  const path = typeof to === "object" ? to.pathname : to;
+  const path = matchPath || (typeof to === "object" ? to.pathname : to);
```

Comment:
I believe we only need to escape the `to` prop. If they give us the `matchPath` prop, we should probably just assume it is already excaped and pass it straight through to `<Route path>`.